### PR TITLE
FIX Don't generate table alias for "from" statement that are not column names

### PR DESCRIPTION
**Pull request by maxime-rainville**
## Manual testing steps
This is updating a low level API that is widely use across tho codebase. There's not really any manual steps that could be taken ... aside from using the system.

## Issues
- #11276

## Pull request checklist
- [x] The target branch is correct
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: tests/php/ORM/SQLSelectTest.php [hunk 1]
```diff
@@ -1327,4 +1327,98 @@ public function testMultipleWithDuplicateName()
 
         $select->addWith('cte', new SQLSelect());
     }
+
+    public function subqueryProvider()
+    {
+        return [
+            'no-alias-string' => ['( SELECT DISTINCT "SQLSelectTest_DO"."ClassName" FROM "SQLSelectTest_DO") AS "FINAL"'],
```
Comment (maxime-rainville) [inline tests/php/ORM/SQLSelectTest.php]: That's the only scenario that would have failed prior to this PR. The other ones are just there for the sake of completeness.

File: tests/php/ORM/SQLSelectTest.php [hunk 2]
```diff
@@ -1327,4 +1327,98 @@ public function testMultipleWithDuplicateName()
...
+
+        $this->assertSQLEquals(
+            'SELECT * FROM ( SELECT DISTINCT "SQLSelectTest_DO"."ClassName" FROM "SQLSelectTest_DO") AS "FINAL"',
+            $actualSQL
+        );
+    }
+
+    public function addFromProvider()
+    {
+        return [
```
Comment (maxime-rainville) [inline tests/php/ORM/SQLSelectTest.php:1358]: Most of those would have pass previously, but `addFrom` didn't have explicit test coverage.

File: src/ORM/Queries/SQLConditionalExpression.php [hunk 3]
```diff
@@ -78,7 +78,18 @@ public function addFrom($from)
         if (is_array($from)) {
             $this->from = array_merge($this->from, $from);
         } elseif (!empty($from)) {
-            $this->from[str_replace(['"','`'], '', $from)] = $from;
+            // Make sure we don't have any pointless blank space in our From clause
+            $from = trim($from);
+
+            // Check if the from clause looks like a regular table name
+            // Table name can contain only alphanumeric characters and underscores
+            if (preg_match('/^["`]?([a-zA-Z0-9_]+)["`]?$/', $from)) {
```
Comment (maxime-rainville) [inline src/ORM/Queries/SQLConditionalExpression.php]: In theory, you could have other characters in a MySQL table name, but we had a pretty strong assumption over the years that only alphanumeric characters and underscores are acceptable.

**Comment by maxime-rainville** [comment]
Got a sink build here with this PR: https://github.com/archiprocode/recipe-kitchen-sink/actions/runs/10103439804

The Lining failure is pre-existing.

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 3 above)
Comment (GuySartorelli) [reply src/ORM/Queries/SQLConditionalExpression.php]: It's probably safer and easier to just say "if there's whitespace, it's not a valid table name. If there's no whitespace, that's all it _can_ be"
```suggestion
            if (preg_match('/[^\s]/', $from)) {
```

File: src/ORM/Queries/SQLConditionalExpression.php [hunk 4]
```diff
@@ -78,7 +78,18 @@ public function addFrom($from)
         if (is_array($from)) {
             $this->from = array_merge($this->from, $from);
         } elseif (!empty($from)) {
-            $this->from[str_replace(['"','`'], '', $from)] = $from;
+            // Make sure we don't have any pointless blank space in our From clause
+            $from = trim($from);
```
Comment (GuySartorelli) [inline src/ORM/Queries/SQLConditionalExpression.php]: This wasn't being done before and doesn't seem related to the bug you're fixing. What's this change for?

File: src/ORM/Queries/SQLConditionalExpression.php [hunk 5]
```diff
@@ -78,7 +78,18 @@ public function addFrom($from)
...
+
+            // Check if the from clause looks like a regular table name
+            // Table name can contain only alphanumeric characters and underscores
+            if (preg_match('/^["`]?([a-zA-Z0-9_]+)["`]?$/', $from)) {
+                // Add an alias for the table name, stripping any quotes
+                $this->from[str_replace(['"','`'], '', $from)] = $from;
+            } else {
+                // Add from clause without an alias
```
Comment (GuySartorelli) [inline src/ORM/Queries/SQLConditionalExpression.php]:
```suggestion
                // Add from clause without an alias - this is probably a full sub-select with its own explicit alias.
```
This seems to be the reasoning - lets leave it here for future generations.

File: tests/php/ORM/SQLSelectTest.php [hunk 6]
```diff
@@ -1327,4 +1327,97 @@ public function testMultipleWithDuplicateName()
 
         $select->addWith('cte', new SQLSelect());
     }
+
+    public function subqueryProvider()
+    {
+        return [
+            'no-alias-string' => ['( SELECT DISTINCT "SQLSelectTest_DO"."ClassName" FROM "SQLSelectTest_DO") AS "FINAL"'],
```
Comment (GuySartorelli) [inline tests/php/ORM/SQLSelectTest.php]: You've called this "no alias string" but the string does explicitly have an alias? It's a bit confusing, but not a deal breaker since it's just a test scenario identifier. I won't hold up the PR merging if this doesn't change.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
I'd love to have `$alias` as a second parameter, so aliases have to be explicitly added instead of the system guessing what it should be.
But that would have to be a major release change so obviously isn't appropriate for this PR.

Seems like a good solution. Just a couple small changes to make.

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 4 above)
Comment (maxime-rainville) [reply src/ORM/Queries/SQLConditionalExpression.php]: I didn't want to trigger the logic below if someone made a call like `$query->addFrom(' MyTableName ')`;

I guess I could chande the regex to something like `^\s*[^\s]+\s*$`. That would match:
- an optional series of spaces followed by
- an non-optional series of none-space followed by 
- an optional series of spaces followed by

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 3 above)
Comment (maxime-rainville) [reply src/ORM/Queries/SQLConditionalExpression.php]: That's a good call.

File: tests/php/ORM/SQLSelectTest.php (hunk 6 above)
Comment (maxime-rainville) [reply tests/php/ORM/SQLSelectTest.php]: I meant to say that the alias wasn't explicitly defined in an array key.

I'll rename it to this.

```suggestion
            'no-explicit-alias-string' => ['( SELECT DISTINCT "SQLSelectTest_DO"."ClassName" FROM "SQLSelectTest_DO") AS "FINAL"'],
```

**Review by maxime-rainville — COMMENTED** [review]
I'll have implement the requested change tomorrow.

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 4 above)
Comment (GuySartorelli) [reply src/ORM/Queries/SQLConditionalExpression.php]: Personally I think if people are putting whitespace around their table name, they should expect errors. But I'm not fussed either way really.

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 3 above)
Comment (maxime-rainville) [reply src/ORM/Queries/SQLConditionalExpression.php]: Removed the trim and tweaked the regex. I added an extra "padded with spaces" test cases. It's worth nothing that the alias will also be padded with spaces which the previous implementation would have done too.

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 4 above)
Comment (maxime-rainville) [reply src/ORM/Queries/SQLConditionalExpression.php]: If it's meant to imitate SQL, you can have as many spaces around your table name as you want. But, I agree it feels a bit yucky allowing spaces there.

File: src/ORM/Queries/SQLConditionalExpression.php (hunk 5 above)
Comment (maxime-rainville) [reply src/ORM/Queries/SQLConditionalExpression.php]: Done.

File: tests/php/ORM/SQLSelectTest.php (hunk 6 above)
Comment (maxime-rainville) [reply tests/php/ORM/SQLSelectTest.php]: Done

File: tests/php/ORM/SQLSelectTest.php [hunk 7]
```diff
@@ -1327,4 +1327,101 @@ public function testMultipleWithDuplicateName()
...
+    }
+
+    public function addFromProvider()
+    {
+        return [
+            'string' => [
+                'MyTable', ['MyTable' => 'MyTable'],
+                'Plain table name get alias automatic alias'
+            ],
+            'string padded with spaces' => [
```
Comment (maxime-rainville) [inline tests/php/ORM/SQLSelectTest.php:1363]: Just added this test case to cover the scenario where the table name is padded with spaces.

**Review by GuySartorelli — APPROVED** [review]
LGTM, thanks for doing this.
